cmd/admin-handlers.go

```diff
@@ -181,8 +181,14 @@ func (adminAPI adminAPIHandlers) ServiceCredentialsHandler(w http.ResponseWriter
 	}
 
 	// Update local credentials in memory.
-	serverConfig.SetCredential(creds)
+	currentCred := serverConfig.SetCredential(creds)
```

Comment:
I think this should be called `prevCred` - the operation will succeed if the lock is acquired. This also makes the error handling for `.Save()` more understandable.